Testscripts/Linux/ntttcp_monitor_throughput.sh

```diff
@@ -0,0 +1,167 @@
+#!/bin/bash
+########################################################################
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+#
+# In this script, we want to monitor the network throughput by ntttcp for running very long time, e.g. one month.
+# Any throughput churn that's upper/lower than (${tolerance} * ${throughput_baseline}) will be logged as Failed (out of tolerance).
+########################################################################
+
+# Source utils.sh
+. utils.sh || {
+    echo "Error: unable to source utils.sh!"
+    echo "TestAborted" > state.txt
+    exit 0
+}
+# Source constants file and initialize most common variables
+UtilsInit
+
+function display_time_from_seconds {
+    local T=$1
+    local D=$((T/60/60/24))
+    local H=$((T/60/60%24))
+    local M=$((T/60%60))
+    local S=$((T%60))
+    (( $D > 0 )) && printf '%d days ' $D
+    (( $H > 0 )) && printf '%d hours ' $H
+    (( $M > 0 )) && printf '%d minutes ' $M
+    (( $D > 0 || $H > 0 || $M > 0 )) && printf 'and '
+    printf '%d seconds\n' $S
+}
+
+function main {
+    while true;do
+        case "$1" in
+            --log_dir)
+                log_dir="$2"
+                shift 2;;
+            --server_internal_ip)
+                server_internal_ip="$2"
+                shift 2;;
+            --) shift; break ;;
+            *) break ;;
+        esac
+    done
+
+    if [[ -z ${seconds_of_runtime} || -z ${monitor_interval} || -z ${tolerance} || -z ${ntttcp_version} ]]; then
+        LogMsg "Error: missing parameter 'seconds_of_runtime' or 'monitor_interval' or 'tolerance' or 'ntttcp_version' "
+        exit 1
+    elif (("${monitor_interval}" > 5)); then
+        LogMsg "Warning: interval larger than 5 seconds may cause testing result not correct"
+    fi
+
+    if [[ -s state.txt ]]; then
+        rm state.txt
+    fi
+
+    for ntttcp_pid in $(pidof ntttcp); do
+        kill $ntttcp_pid
+    done
+
+    if [[ -d "${log_dir}" ]]; then
+        rm -rf "${log_dir}"
+    fi
+    mkdir ${log_dir}
+    if [[ -s "constants.sh" ]]; then
+        cp -rf *.sh ${log_dir}
+    fi
+
+    update_repos
+    install_package "gcc make wget moreutils bc"
+    build_ntttcp ${ntttcp_version}
+
+    # start testing depending on role (10.0.0.4 is receiver/server, other ip is sender/client)
+    private_ip=$(hostname -I | awk '{print $1}')
+    if [[ "${private_ip}" = "${server_internal_ip}" ]]; then
+        nohup ntttcp -r -H &
+        sleep 10
+        SetTestStateRunning
+        echo "Ntttcp_Monitor_Server_Started"
```

Comment:
Why not to use LogMsg instead of echo?
Does it concern double logging? In TestExecution.log and script run output redirected file?

Reply:
thanks, should be a tiny inconsistency, but sure, this will be updated in following PR.